From c14a4b5ee4cd4ab5af97e03b2accff2ee4bd1680 Mon Sep 17 00:00:00 2001
From: Jon Nettleton <jon@laptop.org>
Date: Wed, 8 Jan 2014 07:55:57 +0100
Subject: [PATCH 1197/1199] video: mxc: Hack to fix broken xbmc detection

*WARNING HACK*  This is a terrible workaround to hack the
drivername to work with existing XBMC binaries that check
modalias to match a string to allow it to run on IMX hardware.

This needs to be fixed in userspace.
---
 drivers/video/mxc/mxc_ipuv3_fb.c |    6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/drivers/video/mxc/mxc_ipuv3_fb.c b/drivers/video/mxc/mxc_ipuv3_fb.c
index 49abe8c..0e80e3f 100644
--- a/drivers/video/mxc/mxc_ipuv3_fb.c
+++ b/drivers/video/mxc/mxc_ipuv3_fb.c
@@ -2345,6 +2345,7 @@ static int mxcfb_probe(struct platform_device *pdev)
 	struct fb_info *fbi;
 	struct mxcfb_info *mxcfbi;
 	struct resource *res;
+	char *fix = kmalloc(11, GFP_KERNEL);
 	int ret = 0;
 
 	dev_dbg(&pdev->dev, "%s enter\n", __func__);
@@ -2354,6 +2355,11 @@ static int mxcfb_probe(struct platform_device *pdev)
 		return pdev->id;
 	}
 
+	strlcpy(fix, pdev->name, 11);
+	pr_info("FB name %s\n", pdev->name);
+	pr_info("FB name %s\n", fix);
+	pdev->name = fix;
+
 	plat_data = devm_kzalloc(&pdev->dev, sizeof(struct
 					ipuv3_fb_platform_data), GFP_KERNEL);
 	if (!plat_data)
-- 
1.7.9.5
